lib/cocoapods/generator/xcconfig/xcconfig_helper.rb

```diff
@@ -288,15 +288,23 @@ def self.settings_for_dependent_targets(target, dependent_targets, test_xcconfig
           unless dependent_targets.empty?
             framework_search_paths = []
             library_search_paths = []
+            frameworks = ''
             dependent_targets.each do |dependent_target|
               if dependent_target.requires_frameworks?
                 framework_search_paths << dependent_target.configuration_build_dir(CONFIGURATION_BUILD_DIR_VARIABLE)
+
+                # It seems that a -framework option should always be added when the framework_search_path is updated,
+                # but conservatively only modifying OTHER_LD_FLAGS if there is a static_framework dependent.
+                if dependent_targets.any?(&:static_framework?)
+                  frameworks += %( -framework "#{dependent_target.product_basename}")
+                end
               else
                 library_search_paths << dependent_target.configuration_build_dir(CONFIGURATION_BUILD_DIR_VARIABLE)
               end
             end
             build_settings['FRAMEWORK_SEARCH_PATHS'] = XCConfigHelper.quote(framework_search_paths.uniq)
             build_settings['LIBRARY_SEARCH_PATHS']   = XCConfigHelper.quote(library_search_paths.uniq)
+            build_settings['OTHER_LDFLAGS']          = frameworks
```

Comment:
I think you may want to be invoking `generate_other_ld_flags` instead that actually sets these flags. You should be setting this within `aggregate_xcconfig.rb` instead.

This method here probably needs to be renamed to just specify adding `search_paths_for_dependent_targets` but it doesn't have to be in this PR.

Reply:
interestingly though the `generate_other_ld_flags` only adds the flags for pod targets that require building (have sources included) which in your case I dont think thats true so maybe look into:

```
        def self.generate_vendored_build_settings(aggregate_target, pod_targets, xcconfig)
          pod_targets.each do |pod_target|
            unless pod_target.should_build? && pod_target.requires_frameworks?
              XCConfigHelper.add_settings_for_file_accessors_of_target(aggregate_target, pod_target, xcconfig)
            end
          end
        end
```